Program.cs

```diff
@@ -5,11 +5,91 @@ namespace Calculator
   public static class Program
   {
     private static void Main(string[] args)
+    {
+      PrintWelcomeMessage();
+
+      while (true)
+      {
+        PerformOneCalculation();
+      }
+    }
+
+    private static void PrintWelcomeMessage()
     {
       Console.WriteLine("Welcome to the calculator!");
       Console.WriteLine("==========================");
+    }
+
+    private static void PerformOneCalculation()
+    {
+      var op = AskForOperator();
+      var n = AskForNumberArray(op);
+      var a = CalculateAnswer(op, n);
+
+      Console.WriteLine("The answer is: " + a);
+      Console.WriteLine();
+    }
+
+    private static string AskForOperator()
+    {
+      string[] allowedOperators = { "+", "-", "*", "/" };
+      Console.Write("Please enter the operator: ");
+      string op = Console.ReadLine();
+      return op;
+    }
+
+    private static int[] AskForNumberArray(string op)
+    {
+      Console.Write($"How many numbers do you want to {op}? ");
+
+      int count = int.Parse(Console.ReadLine());
```

Comment:
If user enters a blank value, (or even a alphanumeric) the Parse will fail so should handle the exception and also create a separate method to validate user input for this step, so method only does a single thing.

Reply:
(y)

Reply:
👍